src/foam/nanos/notification/email/DAOResourceLoader.java

```diff
@@ -32,49 +37,34 @@
 {
   protected String groupId_;
   protected String locale_;
+  protected String spid_;
 
-  public static EmailTemplate findTemplate(X x, String name, String groupId, String locale) {
+  public static EmailTemplate findTemplate(X x, String name, String groupId, String locale, String spid, Map templateArgs) {
     DAO groupDAO = (DAO) x.get("groupDAO");
     DAO emailTemplateDAO = (DAO) x.get("localEmailTemplateDAO");
+    EmailTemplate emailTemplate = null;
 
-    /*
-    TODO:
+    do {
+      boolean group_  = ! SafetyUtil.isEmpty(groupId);
+      boolean spid_   = ! SafetyUtil.isEmpty(spid);
 
-    name  group locale spid
```

Comment:
we should leave the matrix as a reference...

Reply:
yes